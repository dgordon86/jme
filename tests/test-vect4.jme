average(a) {
 total = 0;
 for (i = 0 ; i < len(a) ; i = i + 1) {
    total = total + a[i];
  }
  return total / len(a);
}

stddeviation(a) {
   
    newarr = [0, 0, 0, 0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]; 
    
    mean = average(a);
    for(i = 0; i < len(a); i = i+1) {
        newarr[i] = (a[i] - mean) ^ 2;
    }
    
    mean = average(newarr);
    
    return sqrt(mean);
   
}

main()
{

a = [9, 2, 5, 4, 12, 7, 8, 11, 9, 3, 7, 4, 12, 5, 4, 10, 9, 6, 9, 4];

  
print(stddeviation(a));

}
