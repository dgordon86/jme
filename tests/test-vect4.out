2.98328677804